### PR TITLE
test: add chainsaw tests

### PR DESCRIPTION
**Pull request by eddycharly**
This PR adds [chainsaw](https://github.com/kyverno/chainsaw) e2e tests.

## Checklist
* [x] I have signed the CLA
* [ ] I have updated/added any relevant documentation

### CODE REVIEW

**Comment by eddycharly** [comment]
Somehow related https://kyverno.io/blog/2023/12/12/kyverno-chainsaw-the-ultimate-end-to-end-testing-tool/

File: .circleci/config.yml [hunk 1]
```diff
@@ -108,6 +123,19 @@ workflows:
...
           <<: *e2e_configuration
+      - rok8s/kubernetes_e2e_tests:
+          name: "Chainsaw End-To-End Kubernetes 1.23"
+          kind_node_image: "kindest/node:v1.23.13@sha256:ef453bb7c79f0e3caba88d2067d4196f427794086a7d0df8df4f019d5e336b61"
+          <<: *chainsaw_configuration
+      - rok8s/kubernetes_e2e_tests:
+          name: "Chainsaw End-To-End Kubernetes 1.24"
+          kind_node_image: "kindest/node:v1.24.7@sha256:577c630ce8e509131eab1aea12c022190978dd2f745aac5eb1fe65c0807eb315"
+          <<: *chainsaw_configuration
+      - rok8s/kubernetes_e2e_tests:
```
Comment (sudermanjr) [inline .circleci/config.yml:134]: I think we should drop 1.23 and 1.24 and move to 1.25-1.27 instead.

**Review by sudermanjr — CHANGES_REQUESTED** [review]
This looks great. I would be okay with going ahead and removing the existing e2e tests just so we don't flood Circle with tests on the next PR.

File: .circleci/config.yml (hunk 1 above)
Comment (eddycharly) [reply .circleci/config.yml:134]: I agree, maybe in a follow up PR ? This is not related to this change.

**Comment by eddycharly** [comment]
> This looks great. I would be okay with going ahead and removing the existing e2e tests just so we don't flood Circle with tests on the next PR.

@sudermanjr done

File: .circleci/config.yml (hunk 1 above)
Comment (sudermanjr) [reply .circleci/config.yml:108]: That's fine with me

**Review by sudermanjr — APPROVED** [review]

**Comment by sudermanjr** [comment]
LGTM. Thanks!

**Pull request merged**